packages/vite/src/node/plugins/define.ts

```diff
@@ -68,16 +68,18 @@ export function definePlugin(config: ResolvedConfig): Plugin {
     const replacementsKeys = Object.keys(replacements)
     const pattern = replacementsKeys.length
       ? new RegExp(
-          // Do not allow preceding '.', but do allow preceding '...' for spread operations
-          '(?<!(?<!\\.\\.)\\.)\\b(' +
+          // Mustn't be preceded by a char that can be part of an identifier
+          //or a '.' that isn't part of a spread operator
```

Comment:
```suggestion
          // or a '.' that isn't part of a spread operator
```

nit: missing a leading space